baybe/surrogates/gaussian_process/kernel_factory.py

```diff
@@ -0,0 +1,55 @@
+"""Kernel factories for the Gaussian process surrogate."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Protocol, Union
+
+from attrs import define, field
+from attrs.validators import instance_of
+
+from baybe.kernels.base import Kernel
+from baybe.searchspace import SearchSpace
+from baybe.serialization.core import (
+    converter,
+    get_base_structure_hook,
+    unstructure_base,
+)
+from baybe.serialization.mixin import SerialMixin
+
+if TYPE_CHECKING:
+    from torch import Tensor
+
+
+class KernelFactory(Protocol):
+    """A protocol defining the interface expected for kernel factories."""
+
+    def __call__(
+        self, searchspace: SearchSpace, train_x: Tensor, train_y: Tensor
+    ) -> Kernel:
+        """Create a :class:`baybe.kernels.base.Kernel` for the given context."""
```

Comment:
Understand. Do you have another suggestion that would fit the single line? Is it OK if I just write "DOE context" instead of just "context"? Guess that would make it clear and even more precise...

Reply:
changed it as described. please reopen if unhappy